### ISSUE TITLE
Can't export locations

### ISSUE THREAD

**Issue by kalipso-cyber** [owntracks/ios#828]
Hi there,

a while back, I had to kill the server that Owntracks was sending my location data to. So for a couple of months (didn't have much time), it was just collecting the waypoints and not sending them anywhere. After a few weeks/months, my iPhone "offloaded" the app as it hadn't been opened for a while.

Today, I re-downloaded the app, and it still shows the badge counter with ~1400 location points that haven't been transferred to the server (since I killed it). I attempted changing the server's address in the app, upon which I was thankfully prompted that this would result in the loss of all my waypoints. I aborted and created an export, but the file's content amounts to fairly little:

**config.ortw:**

```json
{
  "_type" : "waypoints",
  "waypoints" : [

  ]
}
```

In other words, there aren't any waypoints in there. I'm assuming that they got lost due to the automatic app offloading? Or did I just somehow botch the export and there's still a chance to save those waypoints? Since they still show in the notification badge...

Thanks in advance!

EDIT: Fixed code formatting.

**Comment by jpmens** [comment]
A _waypoint_ is not a _location publish_; what you are exporting are waypoints (aka regions) and since you apparently don't have any configured your export file is essentially empty of them.

Unless you bring up a server at the same domain name or address (depending on how you configured the app), you will lose your 1400 location publishes. However if you "unkill" your server, i.e. bring up an MQTT broker or HTTP endpoint (depending on what you configured), and the app connects to it, it will attempt to publish the 1400 locations which you then will not lose (assuming the backend is correctly set up).

I hope that makes sense.

**Comment by jpmens** [comment]
To clarify: the OwnTracks app badge is showing _location publishes_ and not _waypoints_.

**Comment by kalipso-cyber** [comment]
Thank you for the prompt reply and the explanations! I had the _waypoints_ and _location publishes_ terminology wrong :)

> Unless you bring up a server at the same domain name [...], you will lose your 1400 location publishes. However if you [...] bring up an [...] HTTP endpoint, and the app connects to it, it will attempt to publish the 1400 locations [...]

Just paraphrasing so I'm sure I understand it correctly: My only option to not lose my location publishes is to deploy an application at the previously used address, correct? If so, I'm guessing I would also need to keep the same API key, right? As I cannot change the connection parameters without losing the location publishes.

**Comment by jpmens** [comment]
You have not yet said which type of backend you were using; the question about the "API key" makes me _assume_ you had an HTTP backend hosted somewhere by somebody. If that is the case, and if you currently have an API key embedded in a URL or as part of the authentication between our OwnTracks app and said backend, then the answer is 'yes'.

Many changes to configuration, specifically server addresses, ports, etc. will cause OwnTracks to clear its internal database, as you know from the initial warning you received in app.

**Comment by kalipso-cyber** [comment]
Sorry, yes, your assumption is correct! I was publishing to a `Dawarich` instance running on a server in my network accessed via private IP. So under "Parameters" in the CONNECTION section, I have `http://192.168.1.2:3000/api/v1/ownttracks/points?api_key=abc123...`.

I'lll look into whether I can specify my own API key in Dawarich; if so it's not a problem to spin up another instance under the same IP, have OwnTracks perform the location publishes to it, export the location publishes via Dawarich, merge them with previous exports and spin up a new, permament deployment of Dawarich to send my location publishes to.

Thank you again for your help!

**Comment by jpmens** [comment]
It appears to be configurable as `current_user.api_key`; I assume you'll know where that is configured, as I do not.

Closing for now.

**Comment by nazgul6** [comment]
@kalipso-cyber just facing **the same exact issue** that you described here.

Did you find a way to import those location publishes into dawarich in the end? I can't see a way to manually update the API for the user in question through dawarich GUI, is there a backend way?

**Comment by kalipso-cyber** [comment]
To be honest I haven't gotten around to trying yet. But lest I'm mistaken you should be able to change the URL in the OwnTracks app to point to a URL/IP under which you have a Dawarich instance running and then it would automatically upload the backlog, no?
